### DISCUSSION TITLE
Plan for the first release

### DISCUSSION THREAD

**Discussion by naosense** [apache/pekko#37] [General]
Where to find a clear plan for the first release, as well as the current work schedule? I mean someone like me may want to do something to help, but don't know what to do.

**Comment by Claudenw** [comment]
It might be easiest to subscribe to the mailing lists: dev@pekko.apache.org and users@pekko.apache.org by sending an email to:
dev-subscribe@pekko.apache.org and users-subscribe@pekko.apache.org

**Reply by naosense** [reply]
Thank you, this helps a lot

**Comment by IainHull** [comment]
The mailing lists are very important, but the github milestones might be a useful summary.

https://github.com/apache/incubator-pekko/milestones

**Comment by pjfanning** [comment]
I can think of 2 areas where we might be in a position to get anyone who has an interest to help out.
* build and review the docs
  * `sbt docs/paradoxBrowse`
  * we need to replace every reference to Lightbend and Akka
  * we have a Pekko logo: https://github.com/mdedetrich/akka-apache-project/discussions/15#discussioncomment-3753336
  * URLs with https://doc.akka.io/ should probably be replaced with https://pekko.apache.org/ (despite the fact we don't yet have that site up and running)
  * we might need to delete some stuff that is not relevant to Pekko
  * replace Akka version migration docs (we should recommend users upgrade to latest OSS friendly Akka version before migrating to Pekko)
    * we'll need to document how to migrate from latest OSS friendly Akka version to Pekko
* review the test failures in CI builds (particularly the nightly jobs - the ones that run for PRs and commits are working better)
  *  https://github.com/apache/incubator-pekko/actions

**Reply by seglo** [reply]
I'm taking a look at replacing the akka-paradox plugin and bring the paradox theme inline (for now) as discussed in https://github.com/apache/incubator-pekko/issues/28

https://github.com/apache/incubator-pekko/issues/32

**Comment by bennati** [comment]
Hi all, I'd be interested in knowing when a first release will happen.
Do you have a rough estimate? the milestones page is useful but does not predict a date for the release.
Thank you!

**Reply by pjfanning** [reply]
No estimate can be given. The project is a volunteer project and we are short on volunteers. And there are lots of modules - each module will be released when it is ready- so the modules you want may not be in the early batches.

**Reply by bennati** [reply]
ok, thank you!

**Comment by sam-byng** [comment]
Are we intending the [1.0.0 Milestone](https://github.com/apache/incubator-pekko/milestone/1) in incubator-pekko to contain the complete list of things needed for the first pekko release or just the incubator-pekko module?

[Note: My assumption here is that we need all the repositories to be converted before pekko can be released, however the [above comment](https://github.com/apache/incubator-pekko/discussions/37#discussioncomment-5252656) makes me doubt this.]

AFAICT the backlog for the incubator-pekko (core repo) only is tracked in the [1.0.0 Milestone](https://github.com/apache/incubator-pekko/milestone/1). Then, in the M0 release mailing list thread: [M0 release for incubator-pekko repo](https://lists.apache.org/thread/cx3t7g9rs44tchoz4jytb74sdtlq16zk) there was unresolved discussion of using Github Projects for tracking issues across all of the repositories. So I'm not sure where the work for other repositories (and therefore the whole pekko release?) is being tracked and how we're gathering this all together to get an idea of what will be in PEKKO v1.0.0. 

If we are using Github Project, newcomers/non-committers can't access the [PEKKO GitHub project](https://github.com/orgs/apache/projects/220/views/1 ) at since it is "Private" and are therefore be excluded from getting a high level view on project status.

**Reply by pjfanning** [reply]
I'm afraid that we are looking for developers to work on tasks. When enough tasks are done, we can think about a release. The milestones have plenty of tasks. They are not all the tasks because we are discovering more work that needs to be done as we go along.

The ASF won't let us make the Github Projects public. ASF contributors can see them but they don't add much extra detail.

We will release this repo first and move onto other repos like https://github.com/apache/incubator-pekko-http then. Some work has been on the other repos but the priority is this repo.

https://github.com/apache/incubator-pekko-sbt-paradox is required for us to generate the docs properly, so that is the only other repo that is part of the first release.

I would expect that once this repo has a release, that most of the other repos (like incubator-pekko-http) will be released within a few months.

**Reply by sam-byng** [reply]
Thanks, that makes more sense now. I should have ~1d a week to work on this over the coming weeks, I'll have a look through the issue list.  Initially plan is focussing on tasks that will help other developers get ramped up and contributing more quickly.

Follow up question: 
Do we release the pekko project as a whole package 'bundled together' somehow, or do we release individual modules to the public separately from one another?

**Reply by pjfanning** [reply]
The ASF regards the source code as the release. We will release zip and tgz versions of the source. We will have a download page on pekko.apache.org for this.

We will also publish all the pekko jars associated with this repo to Maven Central - but this is regarded by the ASF as a convenience for users, as opposed to being the release in itself. What gets pushed to Maven Central is very visible in the build.sbt file. If you are not familiar with sbt, you might find it very hard to contribute.

**Comment by Claudenw** [comment]
My observations are as follows:


   1. milestones only apply to a single git repository.  So Pekko should
   use milestones to identify what needs to be done to get to the next
   "release" -- note the quotes, by this I mean the next major thing being
   ready, not necessarily a package release.
   2. projects can span multiple repositories and should be used to track
   package milestones that are needed for an actual project release.  So if
   you need pekko-core and pekko-http to create a release, each should have a
   milestone and those milestones should be referenced in the project for the
   release.  In this way we can track what needs to be completed for a release.
   3. WE NEED VOLUNTEERS.  This should have been #1.  We should be
   accepting of all offers of assistance; writing code, updating
   documentation, creating icons/logos, managing the release, documenting the
   process, testing code, testing documentation, testing process.  If anyone
   wants to help, I am certain that we can find a place to use your talents.

As for me, my talents are not in Scala, but I do know the processes
associated with several Apache processes, and I can help navigate the
Apache way.
Claude
